ui/dashboards/src/components/SaveDashboardButton/SaveDashboardButton.tsx

```diff
@@ -64,18 +72,20 @@ export const SaveDashboardButton = ({ onSave, isDisabled, variant = 'contained'
     }
   };
 
-  const saveDashboard = () => {
-    if (onSave !== undefined) {
-      setSavingDashboard(true);
-      onSave(dashboard)
-        .then(() => {
-          setSavingDashboard(false);
-          closeSaveChangesConfirmationDialog();
-          setEditMode(false);
-        })
-        .catch(() => {
-          setSavingDashboard(false);
-        });
+  const saveDashboard = async () => {
+    if (onSave) {
+      try {
+        setSavingDashboard(true);
+        await onSave(dashboard);
+        setSavingDashboard(false);
+        closeSaveChangesConfirmationDialog();
+        setEditMode(false);
+      } catch (error) {
+        setSavingDashboard(false);
+        if (showErrorSnackbar) {
```

Comment:
This is off by default since exceptionSnackbar logic is handled [here](https://github.com/perses/perses/blob/70bcb3856bc4b53bc3a37abb31d23fcb64ec5aad/ui/app/src/views/ViewDashboard.tsx#L103-L132) in Perses app but other consumers that use gql can opt-in to showing the snackbar here using new optional showErrorSnackbar prop

Reply:
Can we just throw an error here and then let the consumer catch it and handle it however they want? That way the save dashboard button doesn't have a dependency on the snackbar provider

Reply:
Yeah I’ll change this, will just take more refactoring internally, ideally we’d be able to use the whole toolbar and more logic could live there too

Reply:
Can you rethrow the error in the catch block after calling setSavingDashboard?